File: Examples/SnapinsSDKExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/ServiceSDKApplication.java
package com.salesforce.snapinssdkexample;

import android.app.Application;

public class ServiceSDKApplication extends Application {
    private ChatSessionListener chatSessionListener;

    @Override
    public void onCreate() {
        super.onCreate();

        chatSessionListener = new ChatSessionListener(getApplicationContext());
    }

    public ChatSessionListener getChatSessionListener() {
        return chatSessionListener;
    }
}

File: Examples/SnapinsSDKKotlinExample/app/src/main/java/com/salesforce/snapinssdkexample/ServiceSDKApplication.kt
/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * Licensed under the BSD 3-Clause license.
 * For full license text, see LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 */

package com.salesforce.snapinssdkexample

import android.app.Application

class ServiceSDKApplication: Application() {
    lateinit var chatSessionListener: ChatSessionListener

    override fun onCreate() {
        super.onCreate()

        // Optional: Set up live agent chat session listener
        chatSessionListener = ChatSessionListener(applicationContext)
    }
}